include/okapi/impl/device/opticalSensor.hpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "api.h"
+#include "okapi/api/control/controllerInput.hpp"
+#include "okapi/api/filter/passthroughFilter.hpp"
+#include <memory>
+
+namespace okapi {
+enum class OpticalSensorOutput {
+  hue,        ///< The color.
+  saturation, ///< The color's intensity relative to its brightness.
+  brightness  ///< The amount of light.
+};
+
+class OpticalSensor : public ControllerInput<double> {
+  public:
+  /**
+   * An optical sensor on a V5 port.
+   *
+   * ```cpp
+   * auto osHue = OpticalSensor(1);
+   * auto osSat = OpticalSensor(1, OpticalSensorOutput::saturation);
+   * ```
+   *
+   * @param iport The V5 port the device uses.
+   * @param ioutput Which sensor output to return from (@ref okapi::OpticalSensor::get).
+   * @param ifilter The filter to use to filter the sensor output. Only the selected output (via
+   * ``ioutput``) is filtered; the other outputs are untouched.
+   */
+  OpticalSensor(std::uint8_t iport,
+                OpticalSensorOutput ioutput = OpticalSensorOutput::hue,
+                std::unique_ptr<Filter> ifilter = std::make_unique<PassthroughFilter>());
+
+  virtual ~OpticalSensor();
+
+  /**
+   * @return The current filtered value of the selected output (configured by the constructor).
+   */
+  virtual double get();
+
+  /**
+   * @return The current hue value in the range ``[0, 360)``.
+   */
+  double getHue();
+
+  /**
+   * @return The current brightness value in the range ``[0, 1]``.
+   */
+  double getBrightness();
+
+  /**
+   * @return The current saturation value in the range ``[0, 1]``.
+   */
+  double getSaturation();
+
+  /**
+   * @return The PWM value of the white LED in the range ``[0, 100]`` or ``PROS_ERR`` if the
+   * operation failed, setting ``errno``.
+   */
+  int32_t getLedPWM();
+
+  /**
+   * Set the PWM value of the white LED.
+   *
+   * @param value The PWM value in the range ``[0, 100]``.
+   * @return ``1`` if the operation was successful or ``PROS_ERR`` if the operation failed, setting
+   * ``errno``.
+   */
+  int32_t setLedPWM(uint8_t value);
+
+  /**
+   * This is not available if gestures are being detected.
```

Comment:
While this does have a title, it ideally needs to be a little blurb about the method, because the title shows up in the documentation overview like 
![image](https://user-images.githubusercontent.com/16546293/98485560-2bfd5600-21cc-11eb-9589-720aa022095d.png)